model/PermissionProvider.php

```diff
@@ -63,29 +67,67 @@ public function getPermissions(User $user, array $resourceIds) {
         return $rights;
     }
 
-    protected function getStateObjects($stateIds) {
+    protected function getStateObjects($stateIds)
+    {
         $wfmodel = $this->getServiceManager()->get(WorkflowModel::SERVICE_ID);
         $stateCache = array();
         foreach ($stateIds as $stateId) {
             if (!empty($stateId)) {
-                $stateCache[$stateId] = $wfmodel->getState($stateId);
+                $state = $wfmodel->getState($stateId);
```

Comment:
This method `WorkflowState::getState` returns a `WorkflowState` interface, which does dot contain the new public methods we created, violating open/closed principle. 

IMO we should remove the new methods and create the state objects as it was before (or introduce a factory for this).

If we introduce new methods to the interface it will create a breaking change and the backporting wont be possible.